### ISSUE TITLE
Typescriptification cleanup

### ISSUE THREAD

**Issue by tbosch** [angular/angular#2335]
After all sources have been converted to TS, we should clean up our project:

- [x] remove `*.es6`, `*.cjs` from being transpiled
- [ ] remove traceur
- [ ] remove `rtts_assert` subproject and figure out a way to trigger development mode
- [x] publish ts sources on npm and remove folder `atscript` in our npm releases
- [ ] remove `es5build.js` from our npm releases
- [ ] publish `*.d.ts` files in our npm release

**Comment by naomiblack** [comment]
@ttowncompiled please bug @jeffbcross for help if you get stuck on any of these.

**Comment by toedter** [comment]
+1, would you also consider to publish the *.d.ts files within the (alpha) releases at code.angularjs.org?

**Comment by jomeier** [comment]
@toedter Definitely support this idea. @alexeagle put an d.ts file online today even it is for Alpha 22. Would appreciate if there's a way to get the proper d.ts files for each alpha instead of searching hacks in the net. Would make things a lot easier. But they are working on it.

**Comment by alexeagle** [comment]
Yes we will distribute the typings via the expected channels.

**Comment by ttowncompiled** [comment]
PR - #2470

**Comment by ttowncompiled** [comment]
After spending most of the week working on this, I feel that this is an issue much more convoluted than we originally thought. I'm not sure if I have the knowledge or expertise to accomplish this on my own. While it would be really nice to have traceur removed from our build, I'm under the impression that no one is blocked on this issue. If otherwise, please note. Therefore, I would like to postpone until I can receive help from someone who has a much better working knowledge of our build path.

**Comment by jeffbcross** [comment]
I think it'd be good for someone with more familiarity of our build process and its history to pair with @ttowncompiled on this. Maybe it could wait until the week of June 20th (Alpha 29) to pick this back up?

**Comment by naomiblack** [comment]
sounds reasonable. punting to alpha-28 so we can spend some more time on it.

**Comment by naomiblack** [comment]
I'm pushing this to alpha-30. @ttowncompiled has made good progress but we need to make a decision re: output to systemjs or commonjs. Ian will summarize in a doc and we'll review on Monday June 22 at the meeting.

**Comment by IgorMinar** [comment]
something to be aware of when removing traceur is that typescript currently emits incorrect sloppy mode code instead of strict mode code. this is a significant issue that we need to be aware because it changes the behavior of the transpiled code. see https://github.com/Microsoft/TypeScript/issues/3576

**Comment by alexeagle** [comment]
Moved some parts of this:
remove traceur: https://github.com/angular/angular/issues/2829
rtts_assert: https://github.com/angular/angular/issues/2830

Removed a couple items we don't plan to do now.

Otherwise, this is already done.
